src/framework/handlers/esmscript.js

```diff
@@ -0,0 +1,97 @@
+export class ScriptCache {
+    static _scripts = new Map();
```

Comment:
I see a problem with this being static with multiple instances of the application being created - they would share the cache. I think the cache needs to be instantiated per application / per `esm script system` or similar to avoid this, similarly to how current script does.

Reply:
I guess the rationale here is that it's analogous to the browser cache, but I see your point. tbh, I'm not 100% comfortable with using the cache anyway, as if theres' a cache miss (the script wasn't marked as preload) then it doesn't automatically resolve, and requires the users to manually load it. If we we're just using browser cache, then it would fetch from the cache or load it, but either way it would always resolve, which is a lot cleaner.

Reply:
Ok I've made this non static @mvaligursky and cleared on app destroy